addons/mail/static/src/core/public_web/user_status_panel.js

```diff
@@ -0,0 +1,46 @@
+import { Component, useState } from "@odoo/owl";
+
+import { discussSidebarItemsRegistry } from "@mail/core/public_web/discuss_sidebar";
+import { useService } from "@web/core/utils/hooks";
+import { ImStatus } from "../common/im_status";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { rpc } from "@web/core/network/rpc";
+
+/**
+ * @typedef {Object} Props
+ * @extends {Component<Props, Env>}
+ */
+export class UserStatusPanel extends Component {
+    static template = "mail.UserStatusPanel";
+    static components = { ImStatus, Dropdown, DropdownItem };
+    static props = [];
+
+    setup() {
+        super.setup();
+        this.store = useState(useService("mail.store"));
+        this.readableImStatusByCode = {
+            online: "Online",
+            away: "Away",
+            busy: "Do Not Disturb",
+            offline: "Offline",
+        };
+    }
+
+    forceImstatus(status) {
+        rpc("/mail/force_im_status", { status: status === "online" ? null : status });
+        rpc("/discuss/settings/mute", { minutes: status === "busy" ? -1 : false });
```

Comment:
It's strange to make 2 consecutive RPCs, shouldn't it be handled in the same route?

Also it feels strange to try to sync both settings, especially if done just client side like this.

Shouldn't the code using mute actually check for busy too instead?

The current approach will also re-set the "mute" setting to False when removing busy even if mute was already set before.